### DISCUSSION TITLE
Number / Integer checks in Kernel/System/VariableCheck

### DISCUSSION THREAD

**Discussion by stefanhaerter** [RotherOSS/otobo#2771] [General]
I ran across the currently existing number check methods in `Kernel/System/VariableCheck.pm`, which are:

- `IsNumber` (line 225)
- `IsInteger` (line 248)
- `IsPositiveInteger` (line 269)

I think they work fine for standard cases. However, I thought about some ideas of improvement:

- Perhaps enhance `IsNumber` with parameters
  - `OnlyPositive` (suggestion: default `0`)
  - `OnlyNegative` (suggestion: default `0`)
  - `IncludeZero` (suggestion: default `1`)

(when taking this path, we could also think about uniting `IsInteger` and `IsPositiveInteger` to one method with the same parameters).

One thing I see less as a suggestion, but more as an actual bug: All three methods do confirm `999` as a number, but not `+999`. Perhaps this is not very relevant, but semantically I would want to advocate for `+999` being a positive number as well as an integer and a positive integer.

What do you think about this suggestions and opinions?

**Comment by bschmalhofer** [comment]
In Principle yes. But I think that we should move into the direction of the common CPAN modules. There already is _Kernel/cpan-lib/Types/Common/Numeric.pm_ which could be leveraged.

**Comment by stefanhaerter** [comment]
This would also be my preferred solution. I was just unsure if there is any specific reason for sticking to an individual solution in OTOBO or against using existing CPAN modules.

**Comment by bschmalhofer** [comment]
Another question is what to do with leading zeros. IMHO is would be fine to trim those.

**Reply by bschmalhofer** [reply]
See also https://github.com/RotherOSS/otobo/discussions/4033
